### PR TITLE
[REVIEW] java bindings for bitwise shift operators [skip ci]

### PR DESCRIPTION
**Pull request by razajafri**

### CODE REVIEW

File: java/src/main/java/ai/rapids/cudf/BinaryOp.java [hunk 1]
```diff
@@ -40,8 +40,13 @@
   BITWISE_OR(16),
   BITWISE_XOR(17),
   LOGICAL_AND(18),
-  LOGICAL_OR(19);
+  LOGICAL_OR(19),
   //NOT IMPLEMENTED YET COALESCE(20); // x == null ? y : x
+  //NOT IMPLEMENTED YET GENERIC_BINARY(21);
+  SHIFT_LEFT(22), // bitwise shift left (<<)
+  SHIFT_RIGHT(23), // bitwise shift right (>>)
+  SHIFT_RIGHT_UNSINGED(24); // bitwise shift right (>>>)
```
Comment (jlowe) [inline java/src/main/java/ai/rapids/cudf/BinaryOp.java]:
```suggestion
  SHIFT_RIGHT_UNSIGNED(24); // bitwise shift right (>>>)
```

File: java/src/test/java/ai/rapids/cudf/BinaryOpTest.java [hunk 2]
```diff
@@ -35,6 +35,7 @@
   private static final Double[] DOUBLES_2 = new Double[]{10.0, 20.0, 30.0, 40.0, 50.0, 60.0, 100.0};
   private static final Boolean[] BOOLEANS_1 = new Boolean[]{true, true, false, false, null};
   private static final Boolean[] BOOLEANS_2 = new Boolean[]{true, false, true, false, true};
+  private static final int[] SHIFT_BY = new int[]{1, 2, 3, 4, 5, 10, 20};
```
Comment (jlowe) [inline java/src/test/java/ai/rapids/cudf/BinaryOpTest.java]: Ideally these shift values for testing should include some unusual cases like 0, small negative values, and large values both on the positive and negative side.

File: java/src/test/java/ai/rapids/cudf/BinaryOpTest.java [hunk 3]
```diff
@@ -992,4 +993,91 @@ public void testOr() {
       }
     }
   }
+
+  @Test
+  public void testShiftLeft() {
+    try (ColumnVector icv = ColumnVector.fromBoxedInts(INTS_2);
```
Comment (jlowe) [inline java/src/test/java/ai/rapids/cudf/BinaryOpTest.java]: `INTS_2` doesn't contain very many interesting values for this test.  None of the values are negative, zero, or very large.  This applies to the other tests as well.

File: java/src/test/java/ai/rapids/cudf/BinaryOpTest.java (hunk 2 above)
Comment (razajafri) [reply java/src/test/java/ai/rapids/cudf/BinaryOpTest.java]: You are right, these tests are not exhaustive, but mostly to test the plumbing. I can make the change if you still want

**Comment by jlowe** [comment]
Nevermind that last comment, wrong PR window. :man_facepalming: 

Thanks for updating the spelling in the enum, but the references to that enum are still mis-spelled.

**Comment by razajafri** [comment]
@jlowe I have addressed the enum renaming. Let me know if you are ok with this PR

**Comment by razajafri** [comment]
@kkraus14 this is ready to go in when you have time

**Pull request merged**